src/app/todo/components/todoItem/todoItem.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { RouterLink, ActivatedRoute } from '@angular/router';
+import { Observable, of } from 'rxjs';
+import { TaskService } from '../../services/todo.services';
+import { ITask } from '../../models/todo.interface';
+import { switchMap } from 'rxjs/operators';
+import { TaskApiService } from '../../services/task-api.service';
+
+@Component({
+  selector: 'todo-item',
+  standalone: true,
+  templateUrl: './todoItem.component.html',
+  styleUrl: './todoItem.component.scss',
+  imports: [CommonModule, RouterLink],
+})
+export class TodoItemComponent implements OnInit {
+  taskItem: Observable<ITask | null> = of(null);
+
+  constructor(
+    private route: ActivatedRoute,
+    private taskApi: TaskApiService,
+    private service: TaskService
+  ) {}
+
+  deleteTask(id: number) {
+    this.service.deleteTask(id).subscribe({
```

Comment:
here instead of going through multiple services, I saw that you send the ID to TaskService then from TaskService you pass the url and id (which you pass here to TaskService) instead of just directly passing the id to taskServiceApi and using url from constants. Also you don't need to define a separate service for BE calls, you can use the same TaskService to create the methods for endpoint calling :)

Reply:
don't forget to unsubscribe